controllers/clusterurlmonitor/clusterurlmonitor.go

```diff
@@ -0,0 +1,123 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clusterurlmonitor
+
+import (
+	"context"
+	"reflect"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/route-monitor-operator/api/v1alpha1"
+	monitoringv1alpha1 "github.com/openshift/route-monitor-operator/api/v1alpha1"
+	"github.com/openshift/route-monitor-operator/pkg/blackboxexporter"
+	utilreconcile "github.com/openshift/route-monitor-operator/pkg/util/reconcile"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+)
+
+// ClusterUrlMonitorReconciler reconciles a ClusterUrlMonitor object
+type ClusterUrlMonitorReconciler struct {
+	client.Client
+	Log    logr.Logger
+	Scheme *runtime.Scheme
+}
+
+const (
+	FinalizerKey string = "clusterurlmonitor.monitoring.openshift.io/clusterurlmonitorcontroller"
```

Comment:
there is a const package for FinalizerNames

Reply:
What's the advantage of having all the constants in a single package? I think this makes it less flexible as you always need to look into a different file (even package) to figure out the value of a constant, and when moving a package you always need to carve out the constants of the constant package and move them along with it.

Reply:
it's a different one, you think it should be the same string?